velox/exec/WindowPartition.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/exec/RowContainer.h"
+#include "velox/vector/BaseVector.h"
+
+namespace facebook::velox::exec {
+class WindowPartition {
+ public:
+  WindowPartition(
+      const std::vector<exec::RowColumn>& columns,
+      const std::vector<TypePtr>& types,
+      velox::memory::MemoryPool* pool);
+
+  /// Retrieves the column at index 'idx'. The function
+  /// knows the index in the input columns list for the argument.
+  VectorPtr argColumn(vector_size_t idx) const;
```

Comment:
A partition may contain large number of rows. In this case, copying all of them into a single vector may require a lot of memory and take a lot of CPU cycles. What's the use case for this method?

Reply:
Like we discussed, this is indeed a con of our current approach. We will move to an extractColumns API instead. I have updated the code.